### PR TITLE
Issue3106 fix linux

### PR DESCRIPTION
**Pull request by fffw**
@myleshorton Can you have a look?

It fixes a few issues that prevents fresh installed Lantern from starting, on all platforms I think.

### CODE REVIEW

File: installer-resources/chained.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+chainedservers:
```
Comment (fffw) [inline installer-resources/chained.yaml]: Without this line, `len(servers.ChainedServers)` will always be zero when reading from chained.yaml

File: installer-resources/linux/lantern.sh [hunk 2]
```diff
@@ -38,6 +41,7 @@ if [ ! -f $LANTERN_USER_BINARY ]; then
   mkdir -p $LANTERN_USER_DIRECTORY/bin
   cp $LANTERN_SOURCE_BINARY $LANTERN_USER_BINARY
   cp $LANTERN_SOURCE_YAML $LANTERN_USER_DIRECTORY
+  cp $LANTERN_CHAINED_YAML $LANTERN_USER_DIRECTORY
```
Comment (fffw) [inline installer-resources/linux/lantern.sh]: We forgot to copy the new file.

File: installer-resources/chained.yaml (hunk 1 above)
Comment (myleshorton) [inline installer-resources/chained.yaml]: We don't actually use the checked in copy - the copy we built with has that
line.

File: src/github.com/getlantern/flashlight/client/bootstrap.go [hunk 3]
```diff
@@ -147,9 +146,11 @@ func bootstrapPath(fileName string) (string, error) {
 			yamldir = dir + "/../Resources"
 		}
 	} else if runtime.GOOS == "linux" {
-		yamldir = dir
+		yamldir = dir + "/../"
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/client/bootstrap.go]: On linux, executable is started from `~/.lantern/bin`, and the config file is placed at `~/.lantern`

File: src/github.com/getlantern/flashlight/config/config.go [hunk 4]
```diff
@@ -144,6 +144,9 @@ func copyGoodOldConfig(configDir, configPath string) {
 	}
 
 	for _, file := range files {
+		if file.IsDir() {
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/config/config.go]: Skip directories to avoid noise.

File: installer-resources/chained.yaml (hunk 1 above)
Comment (fffw) [inline installer-resources/chained.yaml]: Ok I see. So my PR only fixes the startup issue on linux.

**Comment by myleshorton** [comment]
LGTM!

**Comment by fffw** [comment]
Thanks @myleshorton ! I'll fix the failed tests later.

**Pull request merged**